console/collections/src/kary_merkle_tree/tests/mod.rs

```diff
@@ -170,7 +170,7 @@ fn check_merkle_tree_depth_3_arity_3_padded<LH: LeafHash<Hash = PH::Hash>, PH: P
 
     // Construct the Merkle tree for the given leaves.
     let merkle_tree = KaryMerkleTree::<LH, PH, 3, ARITY>::new(leaf_hasher, path_hasher, &leaves)?;
-    assert_eq!(40, merkle_tree.tree.len());
+    assert_eq!(25, merkle_tree.tree.len());
```

Comment:
With this PR, the nodes that would only contain empty hashes (padding) no longer need to allocate them at the leaf level, which means that the tree ends up being smaller; you can see a calculation of the difference in the current number of leaves - padded for the entire leaf level - and the one this PR proposes (only padded up to the arity) [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=96f2533f23996449cc35364b53b75c79). In case of this particular test case (arity = 3, depth = 3, leaves = 10), the number of tree members that are just padding is currently 17, which this PR takes down to 2, which is a reduction of 15, leading to the corresponding reduction in tree size.

Reply:
Does this change apply to the `merkle-tree` too? (non-k-ary one) If so we shoudl optimize and profile that one too.

Reply:
I've already done that; the wins aren't as great, but I'm happy to file a PR soon if this direction is desirable.

Reply:
https://github.com/AleoHQ/snarkVM/pull/2415